shared/src/commonTest/kotlin/com/mbta/tid/mbta_app/model/NearbyResponseTest.kt

```diff
@@ -2919,4 +2847,281 @@ class NearbyResponseTest {
+                    }
+                }
+            }
+
+        assertEquals(
+            listOf(
+                StopsAssociated.WithRoute(
+                    route,
+                    listOf(
+                        PatternsByStop(
+                            route,
+                            stop,
+                            listOf(
+                                RealtimePatterns.ByHeadsign(
+                                    route,
+                                    "A",
+                                    null,
+                                    listOf(routePattern1),
+                                    listOf(objects.upcomingTrip(sched1))
+                                )
+                            )
+                        )
+                    )
+                )
+            ),
+            staticData.withRealtimeInfo(
+                globalData = GlobalResponse(objects),
+                sortByDistanceFrom = stop.position,
+                schedules = ScheduleResponse(objects),
+                predictions = PredictionsStreamDataResponse(objects),
+                alerts = AlertsStreamDataResponse(objects),
+                filterAtTime = time,
+                pinnedRoutes = setOf(),
+                useTripHeadsigns = anyBoolean(),
+            )
+        )
+    }
+
+    @Test
+    fun `withRealtimeInfo north station disruption case`() {
+        val objects = ObjectCollectionBuilder()
+        val northStation =
+            objects.stop {
+                id = "place-north"
+                locationType = LocationType.STATION
+                childStopIds = listOf("70027", "70026")
+            }
+
+        val northStationNorthboundPlatform =
+            objects.stop {
+                id = "70027"
+                locationType = LocationType.STOP
+                parentStationId = "place-north"
+            }
+
+        val northStationSouthboundPlatform =
+            objects.stop {
+                id = "70026"
+                locationType = LocationType.STOP
+                parentStationId = "place-north"
+            }
+
+        val oakGrove =
+            objects.stop {
+                id = "place-ogmnl"
+                locationType = LocationType.STATION
+                childStopIds = listOf("70036")
+            }
+
+        val oakGrovePlatform =
+            objects.stop {
+                id = "70036"
+                locationType = LocationType.STOP
+                parentStationId = "place-ogmnl"
+            }
+
+        val forestHills =
+            objects.stop {
+                id = "place-forhl"
+                locationType = LocationType.STATION
+                childStopIds = listOf("70001")
+            }
+
+        val forestHillsPlatform =
+            objects.stop {
+                id = "70001"
+                locationType = LocationType.STOP
+                parentStationId = "place-forhl"
+            }
+
```

Comment:
I think the test passes as-is now because isTypical = true, isUpcoming = false, isLastStopOnRoutePattern = true, isArrivalOnly = false (because there are no predictions).

For this test we want to include a prediction order to test the isArrivalOnly piece. 
```

        val northboundPrediction =
            objects.prediction {
                trip = orangeNorthboundTypicalTrip
                stopId = northStationNorthboundPlatform.id
                stopSequence = 130
                arrivalTime = time + 8.minutes
                departureTime = null
            }
```
When that is added, the test fails.